front/components/assistant/conversation/ConversationMessage.tsx

```diff
@@ -1,28 +1,90 @@
-import { Avatar, IconButton } from "@dust-tt/sparkle";
-import { ComponentType, MouseEventHandler } from "react";
+import { Avatar, Button, DropdownMenu } from "@dust-tt/sparkle";
+import { Emoji, EmojiMartData } from "@emoji-mart/data";
+import Picker from "@emoji-mart/react";
+import { ComponentType, MouseEventHandler, useEffect, useState } from "react";
+import React from "react";
+import { mutate } from "swr";
+
+import { classNames } from "@app/lib/utils";
+import { MessageReactionType } from "@app/types/assistant/conversation";
+import { UserType, WorkspaceType } from "@app/types/user";
+
+const MAX_REACTIONS_TO_SHOW = 15;
 
 /**
  * Parent component for both UserMessage and AgentMessage, to ensure avatar,
  * side buttons and spacing are consistent between the two
  */
 export function ConversationMessage({
+  owner,
+  user,
+  conversationId,
   messageId,
   children,
   name,
   pictureUrl,
   buttons,
+  reactions,
   avatarBusy = false,
 }: {
+  owner: WorkspaceType;
+  user: UserType;
+  conversationId: string;
+  messageId: string;
   children?: React.ReactNode;
   name: string | null;
-  messageId: string;
   pictureUrl?: string | null;
   buttons?: {
     icon: ComponentType;
     onClick: MouseEventHandler<HTMLButtonElement>;
   }[];
+  reactions: MessageReactionType[];
   avatarBusy?: boolean;
 }) {
+  const [emojiData, setEmojiData] = useState<EmojiMartData | null>(null);
+
+  useEffect(() => {
+    async function loadEmojiData() {
+      const mod = await import("@emoji-mart/data");
+      const data: EmojiMartData = mod.default as EmojiMartData;
+      setEmojiData(data);
+    }
+
+    void loadEmojiData();
+  }, []);
+
+  const handleSelectEmoji = async ({
```

Comment:
micro nit: `handleEmoji` maybe because Select biases a lot toward the non deletion case?